metricbeat/helper/elastic/elastic.go

```diff
@@ -127,3 +127,36 @@ func FixTimestampField(m common.MapStr, field string) error {
 	}
 	return nil
 }
+
+func ConfigureModule(base mb.Module, xpackEnabledMetricsets []string, register *mb.Register) error {
+	config := struct {
+		XPackEnabled bool `config:"xpack.enabled"`
+	}{}
+	if err := base.UnpackConfig(&config); err != nil {
+		return err
+	}
+
+	// No special configuration is needed if xpack.enabled != true
+	if !config.XPackEnabled {
+		return nil
+	}
+
+	var raw common.MapStr
+	if err := base.UnpackConfig(&raw); err != nil {
+		return err
+	}
+
+	// These metricsets are exactly the ones required if xpack.enabled == true
+	raw["metricsets"] = xpackEnabledMetricsets
+
+	newConfig, err := common.NewConfigFrom(raw)
+	if err != nil {
+		return err
+	}
+
+	if err := base.ReConfigure(newConfig, register); err != nil {
+		return err
```

Comment:
Use `errors.Wrapf` to provide context on the errors of this method.

Reply:
Done in d31485b83.